### PR TITLE
Deterministic js

### PR DESCRIPTION
**Pull request by bebef1987**

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -11,6 +11,9 @@ mozdevice = "*"
 click = "*"
 click-config-file = "*"
 selenium = "*"
+mitmproxy = "==2.0.2"
+cryptography = "==1.8.2"
+beautifulsoup4 = "*"
 
 [requires]
-python_version = "2.7"
+python_version = "3.6"
```
Comment (davehunt) [inline Pipfile:19]: We should do this in a separate patch/pull request.

File: studio.py [hunk 2]
```diff
@@ -16,14 +16,15 @@
 @click.command()
 @click.option("--record/--replay", default=False)
 @click.option("--certutil", required=True, help="Path to certutil.")
+@click.option("--deterministic", required=True, help="Use deterministic JS in the recording")
```
Comment (davehunt) [inline studio.py:20]: Is there a reason not to run with determinism? We can be opinionated here, so I think we should just make this the default and not provide an option to disable it.

File: studio.py [hunk 3]
```diff
@@ -163,8 +168,9 @@ def cli(record, certutil, url, browser_name, browser_path, path):
 
             driver.get(url)
 
-        # wait for mitmdump to finish
-        mitmdump_process.wait()
+        # # wait for mitmdump to finish
+        # mitmdump_process.wait()
+        input("What is your name? ")
```
Comment (davehunt) [inline studio.py:176]: Is this debug?

**Review by davehunt — CHANGES_REQUESTED** [review]

**Comment by bebef1987** [comment]
@davehunt as we have massive changes closing this branch

**Pull request closed without merging**